src/tests/profiler/unittest/releaseonshutdown.cs

```diff
@@ -0,0 +1,66 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Threading;
+
+namespace Profiler.Tests
+{
+    class ReleaseOnShutdown
+    {
+        static readonly Guid ReleaseOnShutdownGuid = new Guid("B8C47A29-9C1D-4EEA-ABA0-8E8B3E3B792E");
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public static void WasteTime()
+        {
+            // Give time for the profiler to detach
+            Console.WriteLine("Sleeping...");
+            for (int i = 0; i < 100; ++i)
+            {
+                Thread.Sleep(100);
+            }
+        }
+
+        public static int RunTest(string[] args)
+        {
+            string profilerName;
+            if (TestLibrary.Utilities.IsWindows)
+            {
+                profilerName = "Profiler.dll";
+            }
+            else if (TestLibrary.Utilities.IsLinux)
+            {
+                profilerName = "libProfiler.so";
+            }
+            else
+            {
+                profilerName = "libProfiler.dylib";
+            }
+
+            string rootPath = Path.GetDirectoryName(System.Reflection.Assembly.GetExecutingAssembly().Location);
+            string profilerPath = Path.Combine(rootPath, profilerName);
+
+            Console.WriteLine($"Attaching profiler {profilerPath} to self.");
+            ProfilerControlHelpers.AttachProfilerToSelf(ReleaseOnShutdownGuid, profilerPath);
+
+            WasteTime();
+
+            return 100;
```

Comment:
I don't see how this ever returns a failure.  Seems like `ReleaseOnShutdown::~ReleaseOnShutdown()` should set some globally observable state on a failure and change the result code.   It could also allow for an early out of the time based sleep if there the detach is complete.

Reply:
This doesn't ever return a failure. Profiler tests are different than normal tests. They relaunch themselves with a profiler attached and then monitor the stdout, in order to pass they must return 100 and have the output contain the phrase "PROFILER TEST PASSES" in stdout.

Most managed profilees will unconditionally return 100, the verification is usually done in the native profiler and it will print the pass message if everything succeeds. The only profilees that don't return 100 are ones that the profiler modifies via rejit or other mechanisms.

Reply:
This code is in the relaunched profiler.  It could easily return a failure code, I understand that it doesn't need to...

It could still be relatively easy for the test to terminate when the destructor sets a process static variable to say the destructor had been run.  This would meet @josalem's concern